### ISSUE TITLE
GUI Side panel detailing information of the selected pixel

### ISSUE THREAD

**Issue by gshorer97** [emilioziniades/flatland#12]
GUI should include a side panel to the right of the canvas

Details to include:
- Square name
- Square current colour
- The X and Y co-ordinates of the current square
- The ownership (Is it claimed/unclaimed and if claimed by whom)
- A button to claim an unclaimed square OR if owned a button to bid on the current square

**Comment by gshorer97** [comment]
Additional features should be included in the side panel if a user selects a square which they own:

- An edit button should appear next to the square name
- An edit button should appear next to the square colour
- Perhaps additional text saying you own this square

**Comment by emilioziniades** [comment]
Great idea dude! That will be really neat, because then all the claiming / bidding functionality will be in the side panel, instead of below the squareManager.

**Comment by gshorer97** [comment]
Agreed, would be nice to have that interface and create changes where the flatland is still in view.

The squareManager is still nice to see all owned squares however

**Comment by gshorer97** [comment]
Thinking that we could use a react card component to display this information

- Card Title: Render square and square name
- Card Subtitle: Renders square owner username OR "Unclaimed square"
- Card Text: Can display coordinates, current colour as a list
- Card Link: Depending on claimed or unclaimed will link to Claim or Bid

Then we can render that area as a 1x2 table with the canvas on the left and info on the right
We can also play with the padding a bit so that the canvas remains more central to the screen

**Comment by gshorer97** [comment]
Discussed to use Alert component instead

**Comment by emilioziniades** [comment]
@gshorer97 Ok, so I think the best way to structure our approach to this app will be to alternate between front-end and smart contracts.

For now, we are taking the smart contract for granted, and building as much functionality as we can with it.

Then we will circle back to the smart contract, make the changes there that we have listed under 0.1 on our projects board, and then proceed to update the front-end and implement the new changes.

And iteratively do that.

I've made some progress on the sidebar so want to update this thread to reflect that. 

Key: ✅ = implemented, ❌ = not implemented, 🤝 = requires changes to the smart contract
Going through the things listed in this thread, the side panel should show:

    - 🤝 Square name
    - ✅  Square current colour
    - ✅ The X and Y co-ordinates of the current square
    - ✅ The ownership (Is it claimed/unclaimed and if claimed by whom)
    - 🤝 A button to claim an unclaimed square OR if owned a button to bid on the current square

    - 🤝 An edit button should appear next to the square name
    - ✅ An edit button should appear next to the square colour (sort of: I've just reused the manageForm component)
    - ✅ Perhaps additional text saying you own this square

Commit 4e3e8feb6de632cb3421d4fbdb97d64790b5c881 has implemented the above. So I am going to close this issue, and reopen a new one with the remaining features which will have to wait until we update the smart contract
